Flow.Launcher/Helper/HotKeyMapper.cs

```diff
@@ -87,14 +72,37 @@ private static bool ShouldIgnoreHotkeys()
             return settings.IgnoreHotkeysOnFullscreen && WindowsInteropHelper.IsWindowFullscreen();
         }
 
+        private static void UpdateLastQUeryMode()
+        {
+            switch(settings.LastQueryMode)
+            {
+                case LastQueryMode.Empty:
+                    mainViewModel.ChangeQueryText(string.Empty);
+                    break;
+                case LastQueryMode.Preserved:
+                    mainViewModel.LastQuerySelected = true;
+                    break;
+                case LastQueryMode.Selected:
+                    mainViewModel.LastQuerySelected = false;
+                    break;
+                default:
+                    throw new ArgumentException($"wrong LastQueryMode: <{settings.LastQueryMode}>");
+
+            }
+        }
+
         private static void SetCustomPluginHotkey()
         {
             if (settings.CustomPluginHotkeys == null) return;
             foreach (CustomPluginHotkey hotkey in settings.CustomPluginHotkeys)
             {
                 SetHotkey(hotkey.Hotkey, (s, e) =>
                 {
-                    if (ShouldIgnoreHotkeys()) return;
+                    if (ShouldIgnoreHotkeys())
+                        return;
+
+                    UpdateLastQUeryMode();
```

Comment:
I don't think we need to add this here. The plugin hotkey is supposed to change the query text to a specific keyword. Therefore, it should always only change the text and allow typing. No need for preserve/select/empty last query.

Reply:
Currently it works for most situation except when the query are the same. Probably we can just add one more check to manually deselect the text.

Reply:
I am still refactoring, maybe take a look when i am done

Reply:
> I don't think we need to add this here. The plugin hotkey is supposed to change the query text to a specific keyword. Therefore, it should always only change the text and allow typing. No need for preserve/select/empty last query.

yeah makes sense, let me double check

Reply:
> > I don't think we need to add this here. The plugin hotkey is supposed to change the query text to a specific keyword. Therefore, it should always only change the text and allow typing. No need for preserve/select/empty last query.
> 
> yeah makes sense, let me double check

removed